### PR TITLE
feat: Views folder migration events from W to R beginning letters

### PR DESCRIPTION
**Pull request by tommasini**
## **Description**
This apply migration of events to Segment from W to R beginning leters of components name under Views folder

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**
Experimetnal Settings: https://recordit.co/Pq9Ck0oyqE
Wallet: https://recordit.co/q4DNzXyTfT
Settings: https://recordit.co/AtfFBbLLwQ

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/RestoreWallet/RestoreWallet.tsx [hunk 1]
```diff
@@ -58,13 +60,13 @@ const RestoreWallet = () => {
...
     );
-  }, [deviceMetaData, previousScreen]);
+  }, [deviceMetaData, previousScreen, trackEvent]);
 
   const handleOnNext = useCallback(async (): Promise<void> => {
     setLoading(true);
     trackEvent(
       MetaMetricsEvents.VAULT_CORRUPTION_RESTORE_WALLET_BUTTON_PRESSED,
-      deviceMetaData,
+      { ...deviceMetaData },
```
Comment (NicolasMassart) [inline app/components/Views/RestoreWallet/RestoreWallet.tsx:69]: Q: Do we really need to spread this?

Comment (tommasini) [reply app/components/Views/RestoreWallet/RestoreWallet.tsx:69]: We need for typescript reasons, the deviceMetadata object is type different than the argument received on the trackEvent  (JsonMap) if you do this you change the ref and you do not have problems anymore, and this doesn't mean that deviceMetaData is wrongly typed I would say

**Review by NicolasMassart — DISMISSED** [review]
Looks good to me.

**Review by NicolasMassart — APPROVED** [review]
Looks good to me.

**Pull request merged**